airflow-core/src/airflow/api_fastapi/core_api/routes/public/hitl.py

```diff
@@ -72,21 +75,31 @@ def _get_task_instance_with_hitl_detail(
     task_id: str,
     session: SessionDep,
     map_index: int,
-) -> TI:
-    query = (
-        select(TI)
-        .where(
-            TI.dag_id == dag_id,
-            TI.run_id == dag_run_id,
-            TI.task_id == task_id,
+    try_number: int | None = None,
+) -> TI | TIH:
+    def _query(orm_object: Base) -> TI | TIH | None:
+        query = (
+            select(orm_object)
+            .where(
+                orm_object.dag_id == dag_id,
+                orm_object.run_id == dag_run_id,
+                orm_object.task_id == task_id,
+                orm_object.map_index == map_index,
+            )
+            .options(joinedload(orm_object.hitl_detail))
         )
-        .options(joinedload(TI.hitl_detail))
-    )
 
-    if map_index is not None:
-        query = query.where(TI.map_index == map_index)
+        if try_number is not None:
+            query = query.where(orm_object.try_number == try_number)
+
+        task_instance = session.scalar(query)
```

Comment:
This an be TIH, we shouldn't name it `task_instance` I believe 
```suggestion
        <other_name> = session.scalar(query)
```

Reply:
This pattern is also used here 

https://github.com/apache/airflow/blob/1aa9b7af38b52afa79d5108a04ceb5fd181682ba/airflow-core/src/airflow/api_fastapi/core_api/routes/public/task_instances.py#L666-L680

This is TI or TIH. both of them are technically task instance. We could renamed it as `ti_or_tih` but I don't think that's better 🤔

Reply:
We can name this `rows`, `entities`, `models` anything not confusing us about having TI while they can be TIH too.

> both of them are technically task instance

I don't understand this statement. Those are not the same model and don't expose the same interface. If I expect a TI and do `get_task_instance` on this I'll have a problem when it's a TIH.

Ideally we could fix both places. That's not a big deal though. (so we can do that latter too, i'm fine)

Reply:
Yep, I think we should rename it on both sides. I'll make the update

Reply:
renamed both places